main.tf

```diff
@@ -75,7 +111,7 @@ resource "ibm_iam_authorization_policy" "kms_policy" {
 }
 
 # workaround for https://github.com/IBM-Cloud/terraform-provider-ibm/issues/4478
-resource "time_sleep" "wait_for_authorization_policy" {
+resource "time_sleep" "wait_for_kms_authorization_policy" {
```

Comment:
it looks like this time sleep was never being used. Can you add it as a depends_on in the `ibm_en_integration` resource